snapshots/devmapper/snapshotter.go

```diff
@@ -398,20 +401,36 @@ func (s *Snapshotter) createSnapshot(ctx context.Context, kind snapshots.Kind, k
 	return mounts, nil
 }
 
-// mkfs creates ext4 filesystem on the given devmapper device
-func mkfs(ctx context.Context, path string) error {
-	args := []string{
-		"-E",
-		// We don't want any zeroing in advance when running mkfs on thin devices (see "man mkfs.ext4")
-		"nodiscard,lazy_itable_init=0,lazy_journal_init=0",
-		path,
+// mkfs creates filesystem on the given devmapper device based on type
+// specified in config. If no file system type is specified in config, ext4
+// is used as the default file system type
+func mkfs(ctx context.Context, fs fsType, path string) error {
+	mkfsCommand := ""
+	var args []string
+
+	switch fs {
+	case fsTypeExt4:
+		mkfsCommand = "mkfs.ext4"
+		args = []string{
+			"-E",
+			// We don't want any zeroing in advance when running mkfs on thin devices (see "man mkfs.ext4")
+			"nodiscard,lazy_itable_init=0,lazy_journal_init=0",
+			path,
+		}
+	case fsTypeXfs:
+		mkfsCommand = "mkfs.xfs"
```

Comment:
Doesn't it write zeros by default?

Reply:
Looks like for mkfs.xfs, it zeroes out only limited size to remove traces of other FS from beginning and end of device. So no non-default option to prevent zeroing seem to exist. 
https://github.com/mtanski/xfsprogs/blob/preadv2/mkfs/xfs_mkfs.c#L2641

Reply:
Using labels now instead of disk reads.